Invalid TS config for `@markdoc/markdoc@0.1.7`

### What happened?

Importing `@markdoc/markdoc` in a fresh TypeScript projects shows multiple errors:

```
node_modules/@markdoc/markdoc/dist/index.d.ts:15:24 - error TS2307: Cannot find module 'markdown-it/lib/token' or its corresponding type declarations.                                                     
                                                                                                                                                                                                           
15 import type Token from 'markdown-it/lib/token';                                                                                                                                                         
                          ~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                          
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/index.d.ts:98:35 - error TS2552: Cannot find name 'Generator'. Did you mean 'Enumerator'?                                                                               
                                                                                                                                                                                                           
98         getAstValues(value: any): Generator<import("./src/types").AstType, void, unknown>;                                                                                                              
                                     ~~~~~~~~~                                                                                                                                                             
                                                                                                                                                                                                           
  node_modules/typescript/lib/lib.scripthost.d.ts:271:13                                                                                                                                                   
    271 declare var Enumerator: EnumeratorConstructor;                                                                                                                                                     
                    ~~~~~~~~~~                                                                                                                                                                             
    'Enumerator' is declared here.                                                                                                                                                                         
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/src/ast/index.d.ts:11:31 - error TS2552: Cannot find name 'Generator'. Did you mean 'Enumerator'?                                                                       
                                                                                                                                                                                                           
11     getAstValues(value: any): Generator<AstType, void, unknown>;                                                                                                                                        
                                 ~~~~~~~~~                                                                                                                                                                 
                                                                                                                                                                                                           
  node_modules/typescript/lib/lib.scripthost.d.ts:271:13                                                                                                                                                   
    271 declare var Enumerator: EnumeratorConstructor;                                                                                                                                                     
                    ~~~~~~~~~~                                                                                                                                                                             
    'Enumerator' is declared here.                                                                                                                                                                         
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/src/ast/node.d.ts:14:13 - error TS2552: Cannot find name 'Generator'. Did you mean 'Enumerator'?                                                                        
                                                                                                                                                                                                           
14     walk(): Generator<Node, void, unknown>;                                                                                                                                                             
               ~~~~~~~~~                                                                                                                                                                                   
                                                                                                                                                                                                           
  node_modules/typescript/lib/lib.scripthost.d.ts:271:13                                                                                                                                                   
    271 declare var Enumerator: EnumeratorConstructor;                                                                                                                                                     
                    ~~~~~~~~~~                                                                                                                                                                             
    'Enumerator' is declared here.                                                                                                                                                                         
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/src/renderers/react/react.d.ts:1:57 - error TS2307: Cannot find module 'react' or its corresponding type declarations.                                                  
                                                                                                                                                                                                           
1 import type { createElement, Fragment, ReactNode } from 'react';                                                                                                                                         
                                                          ~~~~~~~                                                                                                                                          
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/src/tokenizer/index.d.ts:1:24 - error TS2307: Cannot find module 'markdown-it/lib' or its corresponding type declarations.                                              
                                                                                                                                                                                                           
1 import MarkdownIt from 'markdown-it/lib';                                                                                                                                                                
                         ~~~~~~~~~~~~~~~~~                                                                                                                                                                 
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/src/tokenizer/index.d.ts:2:24 - error TS2307: Cannot find module 'markdown-it/lib/token' or its corresponding type declarations.                                        
                                                                                                                                                                                                           
2 import type Token from 'markdown-it/lib/token';                                                                                                                                                          
                         ~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                           
                                                                                                                                                                                                           
node_modules/@markdoc/markdoc/dist/src/utils.d.ts:1:24 - error TS2307: Cannot find module 'markdown-it/lib/token' or its corresponding type declarations.                                                  
                                                                                                     
1 import type Token from 'markdown-it/lib/token';                                                    
                         ~~~~~~~~~~~~~~~~~~~~~~~                                                     
```

Some of these errors are due to missing dependencies `@types/react` and `@types/markdown-it`

### To reproduce

1. Download or clone https://gist.github.com/alunny-stripe/ed7e03f7a6fd712a8ea93eee36af8dbb
2. Run `npm install`
3. Run `npm run types`

### Version

0.1.7

### Additional context

I ran into slightly different errors on my project, which is using TypeScript 4.5.4 and pnpm. Here are the TS@4.5.4 errors:

```
node_modules/@markdoc/markdoc/dist/index.d.ts:15:24 - error TS2307: Cannot find module 'markdown-it/lib/token' or its corresponding type declarations.                                                     

15 import type Token from 'markdown-it/lib/token';
                          ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markdoc/markdoc/dist/index.d.ts:98:35 - error TS2552: Cannot find name 'Generator'. Did you mean 'Enumerator'?

98         getAstValues(value: any): Generator<import("./src/types").AstType, void, unknown>;
                                     ~~~~~~~~~

  node_modules/typescript/lib/lib.scripthost.d.ts:271:13
    271 declare var Enumerator: EnumeratorConstructor;
                    ~~~~~~~~~~
    'Enumerator' is declared here.

node_modules/@markdoc/markdoc/dist/index.d.ts:111:48 - error TS2694: Namespace '"/Users/alunny/dev/markdoc-ts/node_modules/@markdoc/markdoc/dist/src/types"' has no exported member 'Variable'.

111         Variable: typeof import("./src/types").Variable;
                                                   ~~~~~~~~

node_modules/@markdoc/markdoc/dist/src/ast/index.d.ts:11:31 - error TS2552: Cannot find name 'Generator'. Did you mean 'Enumerator'?

11     getAstValues(value: any): Generator<AstType, void, unknown>;
                                 ~~~~~~~~~

  node_modules/typescript/lib/lib.scripthost.d.ts:271:13
    271 declare var Enumerator: EnumeratorConstructor;
                    ~~~~~~~~~~
    'Enumerator' is declared here.

node_modules/@markdoc/markdoc/dist/src/ast/node.d.ts:14:13 - error TS2552: Cannot find name 'Generator'. Did you mean 'Enumerator'?

14     walk(): Generator<Node, void, unknown>;
               ~~~~~~~~~

  node_modules/typescript/lib/lib.scripthost.d.ts:271:13
    271 declare var Enumerator: EnumeratorConstructor;
                    ~~~~~~~~~~
    'Enumerator' is declared here.

node_modules/@markdoc/markdoc/dist/src/renderers/react/react.d.ts:1:57 - error TS2307: Cannot find module 'react' or its corresponding type declarations.

1 import type { createElement, Fragment, ReactNode } from 'react';
                                                          ~~~~~~~

node_modules/@markdoc/markdoc/dist/src/tokenizer/index.d.ts:1:24 - error TS2307: Cannot find module 'markdown-it/lib' or its corresponding type declarations.

1 import MarkdownIt from 'markdown-it/lib';
                         ~~~~~~~~~~~~~~~~~

node_modules/@markdoc/markdoc/dist/src/tokenizer/index.d.ts:2:24 - error TS2307: Cannot find module 'markdown-it/lib/token' or its corresponding type declarations.

2 import type Token from 'markdown-it/lib/token';
                         ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markdoc/markdoc/dist/src/utils.d.ts:1:24 - error TS2307: Cannot find module 'markdown-it/lib/token' or its corresponding type declarations.

1 import type Token from 'markdown-it/lib/token';
```

Thanks @mfix-stripe! With the changes in #197, this is the minimal `tsconfig.json` I need to get things passing in TS 4.8.3:

```json
{
  "compilerOptions": {
    "esModuleInterop": true,
    "moduleResolution": "node",
    "target": "esnext"
  }
}
```

As Ben says, it would be great to add that to the README 😅 

On TS 4.5 (which we use internally) and up to 4.7, I still get the following error:
```
node_modules/@markdoc/markdoc/dist/index.d.ts:111:48 - error TS2694: Namespace '"/Users/alunny/dev/markdoc-ts/node_modules/@markdoc/markdoc/dist/src/types"' has no exported member 'Variable'.

111         Variable: typeof import("./src/types").Variable;                                              ~~~~~~~~
```

I wonder if there was a change in how `esModuleInterop` in TS 4.8 🤷

--- a/README.md
+++ b/README.md
@@ -52,6 +52,19 @@ return Markdoc.renderers.react(content, React);
 
 Check out [our docs](https://markdoc.dev/docs) for more guidance on how to use Markdoc.
 
+### TypeScript
+This is the minimal `tsconfig.json` in order to use Markdoc in your TypeScript project
+
+```json
+{
+  "compilerOptions": {
+    "moduleResolution": "node",
+    "target": "esnext", // works with es2015 or greater
+    "esModuleInterop": true
+  }
+}
+```
+
 ## Contributing
 
 Contributions and feedback are welcome and encouraged. Check out our [contributing guidelines](.github/CONTRIBUTING.md) on how to do so.

--- a/package.json
+++ b/package.json
@@ -34,9 +34,9 @@
   },
   "devDependencies": {
     "@types/jasmine": "^3.10.2",
-    "@types/markdown-it": "^12.2.3",
+    "@types/markdown-it": "*",
     "@types/node": "^16.11.58",
-    "@types/react": "^17.0.49",
+    "@types/react": "*",
     "@typescript-eslint/eslint-plugin": "^5.12.0",
     "@typescript-eslint/parser": "^5.12.0",
     "deep-assert": "0.3.0",
@@ -53,6 +53,10 @@
     "typedoc": "^0.22.13",
     "typescript": "^4.5.4",
     "yaml-js": "^0.2.3"
+  },
+  "peerDependencies": {
+    "@types/markdown-it": "*",
+    "@types/react": "*"
   },
   "repository": {
     "type": "git",

--- a/package.json
+++ b/package.json
@@ -35,8 +35,8 @@
   "devDependencies": {
     "@types/jasmine": "^3.10.2",
     "@types/markdown-it": "^12.2.3",
-    "@types/node": "^16.11.12",
-    "@types/react": "^17.0.38",
+    "@types/node": "^16.11.58",
+    "@types/react": "^17.0.49",
     "@typescript-eslint/eslint-plugin": "^5.12.0",
     "@typescript-eslint/parser": "^5.12.0",
     "deep-assert": "0.3.0",

--- a/package.json
+++ b/package.json
@@ -32,6 +32,10 @@
   "engines": {
     "node": ">=14.7.0"
   },
+  "dependencies": {
+    "@types/markdown-it": "*",
+    "@types/react": "*"
+  },
   "devDependencies": {
     "@types/jasmine": "^3.10.2",
     "@types/markdown-it": "*",
@@ -53,10 +57,6 @@
     "typedoc": "^0.22.13",
     "typescript": "^4.5.4",
     "yaml-js": "^0.2.3"
-  },
-  "peerDependencies": {
-    "@types/markdown-it": "*",
-    "@types/react": "*"
   },
   "repository": {
     "type": "git",

--- a/package.json
+++ b/package.json
@@ -38,9 +38,7 @@
   },
   "devDependencies": {
     "@types/jasmine": "^3.10.2",
-    "@types/markdown-it": "*",
     "@types/node": "^16.11.58",
-    "@types/react": "*",
     "@typescript-eslint/eslint-plugin": "^5.12.0",
     "@typescript-eslint/parser": "^5.12.0",
     "deep-assert": "0.3.0",

